### PR TITLE
Manage Catalogue Solutions e2e tests added

### PR DESCRIPTION
**Pull request by EnitanOO**
AB#14687
AB#15449

### CODE REVIEW

File: tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Actions/Admin/AddSolution.cs [hunk 1]
```diff
@@ -55,6 +56,92 @@ internal bool SaveSolutionButtonDisplayed()
             return ElementDisplayed(Objects.Admin.AddSolutionObjects.SaveSolutionButton);
         }
 
+        public void ManageCatalougeSolution()
```
Comment (bjss-chris) [inline tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Actions/Admin/AddSolution.cs:59]: Typo in catalogue.

File: tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Actions/Admin/AddSolution.cs [hunk 2]
```diff
@@ -55,6 +56,92 @@ internal bool SaveSolutionButtonDisplayed()
...
+        public int NumberOfFilterRadioButtonsDisplayed()
+        {
+            ManageCatalougeSolution();
+            ClickFilterCatalogueSolutionsButton();
+            return Driver.FindElements(Objects.Admin.AddSolutionObjects.FilterRadioButton).Count;
+        }
+
+        public PublicationStatus SelectFilterRadioButton(int index = 0)
+        {
+            Wait.Until(d => NumberOfFilterRadioButtonsDisplayed() > 0);
```
Comment (bjss-chris) [inline tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Actions/Admin/AddSolution.cs:83]: Parameter `d` isn't used so we can use a discard.

File: tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Actions/Admin/AddSolution.cs [hunk 3]
```diff
@@ -55,6 +56,92 @@ internal bool SaveSolutionButtonDisplayed()
...
+            ClickApplyFilterButton();
+            return publicationStatus;
+        }
+
+        public bool AddSolutionLinkDisplayed()
+        {
+            ManageCatalougeSolution();
+            try
+            {
+                Wait.Until(s => ElementDisplayed(Objects.Admin.AddSolutionObjects.AddSolutionLink));
```
Comment (bjss-chris) [inline tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Actions/Admin/AddSolution.cs:103]: Similar to above, parameter `s` isn't used so we can use a discard.

File: tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Actions/Admin/AddSolution.cs [hunk 4]
```diff
@@ -55,6 +56,92 @@ internal bool SaveSolutionButtonDisplayed()
...
+            }
+            catch
+            {
+                return false;
+            }
+        }
+
+        public int GetNumberOfItemsInTable()
+        {
+            int numberOfItems = -1;
```
Comment (bjss-chris) [inline tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Actions/Admin/AddSolution.cs:128]: Initial value is never used so can be removed.

File: tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Actions/Admin/AddSolution.cs [hunk 5]
```diff
@@ -55,6 +56,92 @@ internal bool SaveSolutionButtonDisplayed()
...
+            try
+            {
+                numberOfItems = Driver.FindElement(Objects.Admin.AddSolutionObjects.CatalogueSolutionTable)
+                   .FindElements(By.CssSelector("tbody tr")).Count();
```
Comment (bjss-chris) [inline tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Actions/Admin/AddSolution.cs:132]: We can use the `Count` property here.

File: tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Actions/Admin/AddSolution.cs [hunk 6]
```diff
@@ -66,6 +153,6 @@ private bool ElementDisplayed(By by)
             {
                 return false;
             }
-        }
+        }
```
Comment (bjss-chris) [inline tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Actions/Admin/AddSolution.cs:156]: Unnecessary white space added after brace.

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Admin/Views/Home/Index.cshtml [hunk 7]
```diff
@@ -25,13 +25,14 @@
...
+                
+                    <vc:nhs-action-link url="@Url.Action(
                                         nameof(CatalogueSolutionsController.Index),
                                         typeof(CatalogueSolutionsController).ControllerName())"
-                                    text="Manage Catalogue Solutions" />
+                                        text="Manage Catalogue Solutions" />
+            </div>
             </div>
-        </div>
         <br><br>
```
Comment (bjss-chris) [inline src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Admin/Views/Home/Index.cshtml:36]: The formatting looks like it's gone slightly skew-whiff here. If you have CodeMaid installed recommend running a clean-up.

File: tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Areas/Admin/ManageCatalogueSolutions.cs [hunk 8]
```diff
@@ -0,0 +1,58 @@
+﻿using System.Linq;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Microsoft.EntityFrameworkCore;
+using NHSD.GPIT.BuyingCatalogue.E2ETests.Utils;
+using NHSD.GPIT.BuyingCatalogue.EntityFramework.Ordering.Models;
```
Comment (bjss-chris) [inline tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Areas/Admin/ManageCatalogueSolutions.cs:6]: Is this needed?

File: tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Areas/Admin/ManageCatalogueSolutions.cs [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+using Xunit;
+
+namespace NHSD.GPIT.BuyingCatalogue.E2ETests.Areas.Admin
+{
+    public class ManageCatalogueSolutions : TestBase, IClassFixture<LocalWebApplicationFactory>
```
Comment (bjss-chris) [inline tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Areas/Admin/ManageCatalogueSolutions.cs:11]: `sealed`

File: tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Objects/Admin/AddSolutionObjects.cs [hunk 10]
```diff
@@ -14,5 +14,20 @@ internal static class AddSolutionObjects
         internal static By SaveSolutionButton => By.Id("Submit");
 
         internal static By FoundationSolution => CustomBy.DataTestId("foundation-solution");
+
+        internal static By AddSolutionLink => By.LinkText("Add a solution");
+
+        internal static By CatalogueSolutionLink => By.LinkText("Manage Catalogue Solutions");
+
+        internal static By CatalogueSolutionRowHead => By.ClassName("nhsuk-table__head");
```
Comment (bjss-chris) [inline tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Objects/Admin/AddSolutionObjects.cs:22]: Expecting this to be used in the near future? (It looks like it isn't currently used.)

File: tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Objects/Admin/AddSolutionObjects.cs [hunk 11]
```diff
@@ -14,5 +14,20 @@ internal static class AddSolutionObjects
...
+
+        internal static By CatalogueSolutionTable => CustomBy.DataTestId("manage-catalogue-solution-list");
+
+        internal static By CatalogueSolutionFilter => By.Id("nhsuk-details0");
+
+        internal static By FilterRadioButton => By.ClassName("nhsuk-radios__item");
+
+        internal static By FilterRadioButtonLabel => By.CssSelector("div.nhsuk-radios__item label.nhsuk-radios__label");
```
Comment (bjss-chris) [inline tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Objects/Admin/AddSolutionObjects.cs:30]: Expecting this to be used in the near future? (It looks like it isn't currently used.)

**Review by bjss-chris — CHANGES_REQUESTED** [review]

File: tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Actions/Admin/AddSolution.cs (hunk 1 above)
Comment (EnitanOO) [reply tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Actions/Admin/AddSolution.cs]: corrected

File: tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Actions/Admin/AddSolution.cs (hunk 2 above)
Comment (EnitanOO) [reply tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Actions/Admin/AddSolution.cs]: done

File: tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Actions/Admin/AddSolution.cs (hunk 3 above)
Comment (EnitanOO) [reply tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Actions/Admin/AddSolution.cs]: Done

File: tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Actions/Admin/AddSolution.cs (hunk 4 above)
Comment (EnitanOO) [reply tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Actions/Admin/AddSolution.cs]: removed

File: tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Actions/Admin/AddSolution.cs (hunk 5 above)
Comment (EnitanOO) [reply tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Actions/Admin/AddSolution.cs]: done

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Admin/Views/Home/Index.cshtml (hunk 7 above)
Comment (EnitanOO) [reply src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Admin/Views/Home/Index.cshtml]: CodeMaid installed and used

File: tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Areas/Admin/ManageCatalogueSolutions.cs (hunk 8 above)
Comment (EnitanOO) [reply tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Areas/Admin/ManageCatalogueSolutions.cs]: removed

File: tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Areas/Admin/ManageCatalogueSolutions.cs (hunk 9 above)
Comment (EnitanOO) [reply tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Areas/Admin/ManageCatalogueSolutions.cs]: done

File: tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Objects/Admin/AddSolutionObjects.cs (hunk 10 above)
Comment (EnitanOO) [reply tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Objects/Admin/AddSolutionObjects.cs]: removed

File: tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Objects/Admin/AddSolutionObjects.cs (hunk 11 above)
Comment (EnitanOO) [reply tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Objects/Admin/AddSolutionObjects.cs]: removed

File: tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Actions/Admin/AddSolution.cs (hunk 6 above)
Comment (EnitanOO) [reply tests/NHSD.GPIT.BuyingCatalogue.E2E.PublicBrowseTests/Actions/Admin/AddSolution.cs:140]: sorted

**Review by bjss-chris — COMMENTED** [review]
Looks fine. Just needs bringing up-to-date with the latest main.

**Review by bjss-chris — APPROVED** [review]
Squash and merge please and can you rewrite your commit message in the imperative?

E.g. 'Add manage catalogue solutions E2E tests'.

I'll expand on some of this stuff when I run the git session.

**Comment by EnitanOO** [comment]
Ok Chris. Thanks!

But I think there has been another change now and not able to do a merge...

**Comment by bjss-chris** [comment]
> Ok Chris. Thanks! But I think there has been another change now and not able to do a merge...

It was fine, no issues. I've just merged it.

**Comment by EnitanOO** [comment]
Thank you Chris

**Pull request merged**